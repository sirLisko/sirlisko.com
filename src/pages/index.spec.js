import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import Index from "./index.js";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `I am the title`
        }
      }
    })
  );
});

describe("Index Page", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = renderer.create(<Index />).toJSON();
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
