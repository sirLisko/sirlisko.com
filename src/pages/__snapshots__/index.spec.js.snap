// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should render properly 1`] = `
<div
  className="home"
>
  <section
    className="life"
  >
    <span
      className="life__heart"
    />
    <span
      className="life__heart"
    />
    <span
      className="life__heart"
    />
  </section>
  <figure
    className="ghost"
    onFocus={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "left": "-155px",
        "top": "-155px",
      }
    }
  />
  <header>
    <ul
      className="links"
    >
      <li
        className="links--blog"
      >
        <a
          href="https://sirlisko.com/blog"
        >
          <p
            className="caption"
          >
            blog
          </p>
        </a>
      </li>
      <li
        className="links--twitter"
      >
        <a
          href="https://twitter.com/sirLisko"
        >
          <p
            className="caption"
          >
            twitter
          </p>
        </a>
      </li>
      <li
        className="links--facebook"
      >
        <a
          href="https://www.facebook.com/sirLisko"
        >
          <p
            className="caption"
          >
            facebook
          </p>
        </a>
      </li>
      <li
        className="links--tumblr"
      >
        <a
          href="https://sirLisko.tumblr.com"
        >
          <p
            className="caption"
          >
            tumblr
          </p>
        </a>
      </li>
      <li
        className="links--github"
      >
        <a
          href="https://github.com/sirLisko"
        >
          <p
            className="caption"
          >
            github
          </p>
        </a>
      </li>
      <li
        className="links--instagram"
      >
        <a
          href="https://instagram.com/sirlisko"
        >
          <p
            className="caption"
          >
            instagram
          </p>
        </a>
      </li>
      <li
        className="links--lastfm"
      >
        <a
          href="https://lastfm.com/user/sirlisko"
        >
          <p
            className="caption"
          >
            lastfm
          </p>
        </a>
      </li>
      <li
        className="links--mail"
      >
        <a
          href="mailto:luca@sirlisko.com"
        >
          <p
            className="caption"
          >
            mail me!
          </p>
        </a>
      </li>
    </ul>
  </header>
  <article>
    <section
      className="greeting"
    >
      <img
        alt="sirlisko face"
        className="greeting__face"
        src="./images/sirlisko.svg"
      />
      <div
        className="greeting__intro"
      >
        <p>
          Ciao!
        </p>
        <p>
          I’m sirLisko,
        </p>
        <p>
          another f@$#&n’
        </p>
        <p
          className="greeting__intro__desc"
        >
          <span
            className="greeting__intro--active"
          >
            web developer
          </span>
          <span
            className=""
          >
            super hero
          </span>
          <span
            className=""
          >
            dreamer
          </span>
          <span
            className=""
          >
            giagguaro
          </span>
          <span
            className=""
          >
            bored guy
          </span>
          <span
            className=""
          >
            8-bit lover
          </span>
          <span
            className=""
          >
            sleeper
          </span>
          <span
            className=""
          >
            bug creator
          </span>
          <span
            className=""
          >
            yak shaver
          </span>
          <span
            className=""
          >
            rubber duck
          </span>
          <span
            className=""
          >
            zombie coder
          </span>
          <span
            className=""
          >
            pizza eater
          </span>
          <span
            className=""
          >
            londoner
          </span>
        </p>
      </div>
    </section>
  </article>
  <footer>
    <a
      className="my__projects"
      href="/projects"
    >
      Projects
    </a>
    <a
      className="my__resume"
      href="/resume"
    >
      Resume
    </a>
  </footer>
</div>
`;
